pkg/operator/controller/ingress/deployment_test.go

```diff
@@ -2265,3 +2265,37 @@ func TestDesiredRouterDeploymentDefaultPlacement(t *testing.T) {
 	}
 
 }
+
+func TestDesiredRouterDeploymentRouterExternalCertificate(t *testing.T) {
+	ic, ingressConfig, infraConfig, apiConfig, networkConfig, _, clusterProxyConfig := getRouterDeploymentComponents(t)
+
+	deployment, err := desiredRouterDeployment(ic, ingressControllerImage, ingressConfig, infraConfig, apiConfig, networkConfig, false, false, nil, clusterProxyConfig, false)
+	if err != nil {
+		t.Fatalf("invalid router Deployment: %v", err)
+	}
+
+	// Verify that router external certificate env var is not added.
+	expectedEnv := []envData{
+		{"ROUTER_EXTERNAL_CERTIFICATE", false, ""},
+	}
+
+	if err := checkDeploymentEnvironment(t, deployment, expectedEnv); err != nil {
```

Comment:
This part could be integrated into the existing `Test_desiredRouterDeployment` just by adding `[]envData{{"ROUTER_EXTERNAL_CERTIFICATE", false, ""}}` to the existing set of env data. That would save us some lines of code and some CPU cycles. But I'm not insisting on this one.

Reply:
I didn't want to tinker with the existing `Test_desiredRouterDeployment()` function as there are other `TestDesiredRouterDeployment__()` functions and I thought each function was designed for testing specific parameters.

Reply:
> I thought each function was designed for testing specific parameters.

Yes. I'm not a fan of how it's done currently but that is a subject for the future refactoring, out of the scope of this PR.

Reply:
I'll then add the new test function. Whenever a future refactoring is taken up then it probably can be combined with other env var test functions.

Reply:
> I'll then add the new test function.

You already did. I'm fine with it.